In [2]:
import os
os.getcwd()

'C:\\Users\\Lenovo'

In [3]:
os.chdir("D://Master//10. TFM//Data")

In [4]:
import pandas as pd
import numpy as np
from sklearn.preprocessing import LabelEncoder, StandardScaler
import matplotlib.pyplot as plt
import seaborn as sns

### Vamos a cargar todos los datasets del e-Commerce

#### Customers

In [5]:
customers = pd.read_csv("customers.csv")

In [6]:
customers

customer_id                customer_unique_id  \
0      06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1      18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2      4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3      b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4      4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   
...                                 ...                               ...   
99436  17ddf5dd5d51696bb3d7c6291687be6f  1a29b476fee25c95fbafc67c5ac95cf8   
99437  e7b71a9017aa05c9a7fd292d714858e8  d52a67c98be1cf6a5c84435bd38d095d   
99438  5e28dfe12db7fb50a4b2f691faecea5e  e9f50caf99f032f0bf3c55141f019d99   
99439  56b18e2166679b8a959d72dd06da27f9  73c2643a0a458b49f58cea58833b192e   
99440  274fa6071e5e17fe303b9748641082c8  84732c5050c01db9b23e19ba39899398   

       customer_zip_code_prefix          customer_city customer_state  
0                         14409                 franca             SP  
1                          9790  sao bernardo do campo             SP  
2                          1151              sao paulo             SP  
3                          8775        mogi das cruzes             SP  
4                         13056               campinas             SP  
...                         ...                    ...            ...  
99436                      3937              sao paulo             SP  
99437                      6764        taboao da serra             SP  
99438                     60115              fortaleza             CE  
99439                     92120                 canoas             RS  
99440                      6703                  cotia             SP  

[99441 rows x 5 columns]

#### Geolocation

In [7]:
geolocation = pd.read_csv("geolocation.csv")

In [8]:
geolocation

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                               1037       -23.545621       -46.639292   
1                               1046       -23.546081       -46.644820   
2                               1046       -23.546129       -46.642951   
3                               1041       -23.544392       -46.639499   
4                               1035       -23.541578       -46.641607   
...                              ...              ...              ...   
1000158                        99950       -28.068639       -52.010705   
1000159                        99900       -27.877125       -52.224882   
1000160                        99950       -28.071855       -52.014716   
1000161                        99980       -28.388932       -51.846871   
1000162                        99950       -28.070104       -52.018658   

        geolocation_city geolocation_state  
0              sao paulo                SP  
1              sao paulo                SP  
2              sao paulo                SP  
3              sao paulo                SP  
4              sao paulo                SP  
...                  ...               ...  
1000158         tapejara                RS  
1000159   getulio vargas                RS  
1000160         tapejara                RS  
1000161  david canabarro                RS  
1000162         tapejara                RS  

[1000163 rows x 5 columns]

#### Order_items

In [9]:
order_items = pd.read_csv("order_items.csv")

In [10]:
order_items

order_id  order_item_id  \
0       00010242fe8c5a6d1ba2dd792cb16214              1   
1       00018f77f2f0320c557190d7a144bdd3              1   
2       000229ec398224ef6ca0657da4fc703e              1   
3       00024acbcdf0a6daa1e931b038114c75              1   
4       00042b26cf59d7ce69dfabb4e55b4fd9              1   
...                                  ...            ...   
112645  fffc94f6ce00a00581880bf54a75a037              1   
112646  fffcd46ef2263f404302a634eb57f7eb              1   
112647  fffce4705a9662cd70adb13d4a31832d              1   
112648  fffe18544ffabc95dfada21779c9644f              1   
112649  fffe41c64501cc87c801fd61db3f6244              1   

                              product_id                         seller_id  \
0       4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1       e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2       c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3       7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4       ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   
...                                  ...                               ...   
112645  4aa6014eceb682077f9dc4bffebc05b0  b8bc237ba3788b23da09c0f1f3a3288c   
112646  32e07fd915822b0765e448c4dd74c828  f3c38ab652836d21de61fb8314b69182   
112647  72a30483855e2eafc67aee5dc2560482  c3cfdc648177fdbbbb35635a37472c53   
112648  9c422a519119dcad7575db5af1ba540e  2b3e4a2a3ea8e01938cabda2a3e5cc79   
112649  350688d9dc1e75ff97be326363655e01  f7ccf836d21b2fb1de37564105216cc1   

        shipping_limit_date   price  freight_value  
0       2017-09-19 09:45:35   58.90          13.29  
1       2017-05-03 11:05:13  239.90          19.93  
2       2018-01-18 14:48:30  199.00          17.87  
3       2018-08-15 10:10:18   12.99          12.79  
4       2017-02-13 13:57:51  199.90          18.14  
...                     ...     ...            ...  
112645  2018-05-02 04:11:01  299.99          43.41  
112646  2018-07-20 04:31:48  350.00          36.53  
112647  2017-10-30 17:14:25   99.90          16.95  
112648  2017-08-21 00:04:32   55.99           8.72  
112649  2018-06-12 17:10:13   43.00          12.79  

[112650 rows x 7 columns]

#### Order_payments

In [11]:
order_payments = pd.read_csv("order_payments.csv")

In [12]:
order_payments

order_id  payment_sequential payment_type  \
0       b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1       a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2       25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3       ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4       42fdf880ba16b47b59251dd489d4441a                   1  credit_card   
...                                  ...                 ...          ...   
103881  0406037ad97740d563a178ecc7a2075c                   1       boleto   
103882  7b905861d7c825891d6347454ea7863f                   1  credit_card   
103883  32609bbb3dd69b3c066a6860554a77bf                   1  credit_card   
103884  b8b61059626efa996a60be9bb9320e10                   1  credit_card   
103885  28bbae6599b09d39ca406b747b6632b1                   1       boleto   

        payment_installments  payment_value  
0                          8          99.33  
1                          1          24.39  
2                          1          65.71  
3                          8         107.78  
4                          2         128.45  
...                      ...            ...  
103881                     1         363.31  
103882                     2          96.80  
103883                     1          47.77  
103884                     5         369.54  
103885                     1         191.58  

[103886 rows x 5 columns]

#### Order_reviews

In [13]:
order_reviews = pd.read_csv("order_reviews.csv")

In [14]:
order_reviews

review_id                          order_id  \
0      7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1      80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2      228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3      e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4      f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   
...                                 ...                               ...   
99219  574ed12dd733e5fa530cfd4bbf39d7c9  2a8c23fee101d4d5662fa670396eb8da   
99220  f3897127253a9592a73be9bdfdf4ed7a  22ec9f0669f784db00fa86d035cf8602   
99221  b3de70c89b1510c4cd3d0649fd302472  55d4004744368f5571d1f590031933e4   
99222  1adeb9d84d72fe4e337617733eb85149  7725825d039fc1f0ceb7635e3f7d9206   
99223  efe49f1d6f951dd88b51e6ccd4cc548f  90531360ecb1eec2a1fbb265a0db0508   

       review_score review_comment_title  \
0                 4                  NaN   
1                 5                  NaN   
2                 5                  NaN   
3                 5                  NaN   
4                 5                  NaN   
...             ...                  ...   
99219             5                  NaN   
99220             5                  NaN   
99221             5                  NaN   
99222             4                  NaN   
99223             1                  NaN   

                                  review_comment_message review_creation_date  \
0                                                    NaN  2018-01-18 00:00:00   
1                                                    NaN  2018-03-10 00:00:00   
2                                                    NaN  2018-02-17 00:00:00   
3                  Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4      Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   
...                                                  ...                  ...   
99219                                                NaN  2018-07-07 00:00:00   
99220                                                NaN  2017-12-09 00:00:00   
99221  Excelente mochila, entrega super rápida. Super...  2018-03-22 00:00:00   
99222                                                NaN  2018-07-01 00:00:00   
99223  meu produto chegou e ja tenho que devolver, po...  2017-07-03 00:00:00   

      review_answer_timestamp  
0         2018-01-18 21:46:59  
1         2018-03-11 03:05:13  
2         2018-02-18 14:36:24  
3         2017-04-21 22:02:06  
4         2018-03-02 10:26:53  
...                       ...  
99219     2018-07-14 17:18:30  
99220     2017-12-11 20:06:42  
99221     2018-03-23 09:10:43  
99222     2018-07-02 12:59:13  
99223     2017-07-03 21:01:49  

[99224 rows x 7 columns]

#### Orders

In [15]:
orders = pd.read_csv("orders.csv")

In [16]:
orders

order_id                       customer_id  \
0      e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1      53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2      47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3      949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4      ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   
...                                 ...                               ...   
99436  9c5dedf39a927c1b2549525ed64a053c  39bd1228ee8140590ac3aca26f2dfe00   
99437  63943bddc261676b46f01ca7ac2f7bd8  1fca14ff2861355f6e5f14306ff977a7   
99438  83c1379a015df1e13d02aae0204711ab  1aa71eb042121263aafbe80c1b562c9c   
99439  11c177c8e97725db2631073c19f07b62  b331b74b18dc79bcdf6532d51e1637c1   
99440  66dea50a8b16d9b4dee7af250b4be1a5  edb027a75a1449115f6b43211ae02a24   

      order_status order_purchase_timestamp    order_approved_at  \
0        delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1        delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2        delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3        delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4        delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   
...            ...                      ...                  ...   
99436    delivered      2017-03-09 09:54:05  2017-03-09 09:54:05   
99437    delivered      2018-02-06 12:58:58  2018-02-06 13:10:37   
99438    delivered      2017-08-27 14:46:43  2017-08-27 15:04:16   
99439    delivered      2018-01-08 21:28:27  2018-01-08 21:36:21   
99440    delivered      2018-03-08 20:57:30  2018-03-09 11:20:28   

      order_delivered_carrier_date order_delivered_customer_date  \
0              2017-10-04 19:55:00           2017-10-10 21:25:13   
1              2018-07-26 14:31:00           2018-08-07 15:27:45   
2              2018-08-08 13:50:00           2018-08-17 18:06:29   
3              2017-11-22 13:39:59           2017-12-02 00:28:42   
4              2018-02-14 19:46:34           2018-02-16 18:17:02   
...                            ...                           ...   
99436          2017-03-10 11:18:03           2017-03-17 15:08:01   
99437          2018-02-07 23:22:42           2018-02-28 17:37:56   
99438          2017-08-28 20:52:26           2017-09-21 11:24:17   
99439          2018-01-12 15:35:03           2018-01-25 23:32:54   
99440          2018-03-09 22:11:59           2018-03-16 13:08:30   

      order_estimated_delivery_date  
0               2017-10-18 00:00:00  
1               2018-08-13 00:00:00  
2               2018-09-04 00:00:00  
3               2017-12-15 00:00:00  
4               2018-02-26 00:00:00  
...                             ...  
99436           2017-03-28 00:00:00  
99437           2018-03-02 00:00:00  
99438           2017-09-27 00:00:00  
99439           2018-02-15 00:00:00  
99440           2018-04-03 00:00:00  

[99441 rows x 8 columns]

#### Products

In [17]:
products = pd.read_csv("products.csv")

In [18]:
products

product_id              product_category_name  \
0      1e9e8ef04dbcff4541ed26657ea517e5                         perfumaria   
1      3aa071139cb16b67ca9e5dea641aaa2f                              artes   
2      96bd76ec8810374ed1b65e291975717f                      esporte_lazer   
3      cef67bcfe19066a932b7673e239eb23d                              bebes   
4      9dc1a7de274444849c219cff195d0b71              utilidades_domesticas   
...                                 ...                                ...   
32946  a0b7d5a992ccda646f2d34e418fff5a0                   moveis_decoracao   
32947  bf4538d88321d0fd4412a93c974510e6  construcao_ferramentas_iluminacao   
32948  9a7c6041fa9592d9d9ef6cfe62a71f8c                    cama_mesa_banho   
32949  83808703fc0706a22e264b9d75f04a2e             informatica_acessorios   
32950  106392145fca363410d287a815be6de4                    cama_mesa_banho   

       product_name_lenght  product_description_lenght  product_photos_qty  \
0                     40.0                       287.0                 1.0   
1                     44.0                       276.0                 1.0   
2                     46.0                       250.0                 1.0   
3                     27.0                       261.0                 1.0   
4                     37.0                       402.0                 4.0   
...                    ...                         ...                 ...   
32946                 45.0                        67.0                 2.0   
32947                 41.0                       971.0                 1.0   
32948                 50.0                       799.0                 1.0   
32949                 60.0                       156.0                 2.0   
32950                 58.0                       309.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
0                 225.0               16.0               10.0   
1                1000.0               30.0               18.0   
2                 154.0               18.0                9.0   
3                 371.0               26.0                4.0   
4                 625.0               20.0               17.0   
...                 ...                ...                ...   
32946           12300.0               40.0               40.0   
32947            1700.0               16.0               19.0   
32948            1400.0               27.0                7.0   
32949             700.0               31.0               13.0   
32950            2083.0               12.0                2.0   

       product_width_cm  
0                  14.0  
1                  20.0  
2                  15.0  
3                  26.0  
4                  13.0  
...                 ...  
32946              40.0  
32947              16.0  
32948              27.0  
32949              20.0  
32950               7.0  

[32951 rows x 9 columns]

#### Sellers

In [19]:
sellers = pd.read_csv("sellers.csv")

In [20]:
sellers

seller_id  seller_zip_code_prefix  \
0     3442f8959a84dea7ee197c632cb2df15                   13023   
1     d1b65fc7debc3361ea86b5f14c68d2e2                   13844   
2     ce3ad9de960102d0677a81f5d0bb7b2d                   20031   
3     c0f3eea2e14555b6faeea3dd58c1b1c3                    4195   
4     51a04a8a6bdcb23deccc82b0b80742cf                   12914   
...                                ...                     ...   
3090  98dddbc4601dd4443ca174359b237166                   87111   
3091  f8201cab383e484733266d1906e2fdfa                   88137   
3092  74871d19219c7d518d0090283e03c137                    4650   
3093  e603cf3fec55f8697c9059638d6c8eb5                   96080   
3094  9e25199f6ef7e7c347120ff175652c3b                   12051   

            seller_city seller_state  
0              campinas           SP  
1            mogi guacu           SP  
2        rio de janeiro           RJ  
3             sao paulo           SP  
4     braganca paulista           SP  
...                 ...          ...  
3090            sarandi           PR  
3091            palhoca           SC  
3092          sao paulo           SP  
3093            pelotas           RS  
3094            taubate           SP  

[3095 rows x 4 columns]

Observamos que hay varios campos interesantes en las diferentes tablas para poder formar un único dataset con datos de pedidos y clientes.

- customer_id
- customer_zip_code_prefix	
- customer_city	
- customer_state
- order_id
- price
- freight_value
- payment_type
- payment_value (debe ser igual a price + freight_value)
- product_id (aquí puede haber varios id dentro de un mismo pedido pero teóricamente podemos sacarlos de la tabla order_items)
- product_category_name	
- order_status	
- order_purchase_timestamp
- order_approved_at
- review_id
- review_score

## Fusión Dataframes

In [21]:
df_0 = pd.merge(customers,orders, on="customer_id", how="inner") 
df_0

customer_id                customer_unique_id  \
0      06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1      18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2      4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3      b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4      4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   
...                                 ...                               ...   
99436  17ddf5dd5d51696bb3d7c6291687be6f  1a29b476fee25c95fbafc67c5ac95cf8   
99437  e7b71a9017aa05c9a7fd292d714858e8  d52a67c98be1cf6a5c84435bd38d095d   
99438  5e28dfe12db7fb50a4b2f691faecea5e  e9f50caf99f032f0bf3c55141f019d99   
99439  56b18e2166679b8a959d72dd06da27f9  73c2643a0a458b49f58cea58833b192e   
99440  274fa6071e5e17fe303b9748641082c8  84732c5050c01db9b23e19ba39899398   

       customer_zip_code_prefix          customer_city customer_state  \
0                         14409                 franca             SP   
1                          9790  sao bernardo do campo             SP   
2                          1151              sao paulo             SP   
3                          8775        mogi das cruzes             SP   
4                         13056               campinas             SP   
...                         ...                    ...            ...   
99436                      3937              sao paulo             SP   
99437                      6764        taboao da serra             SP   
99438                     60115              fortaleza             CE   
99439                     92120                 canoas             RS   
99440                      6703                  cotia             SP   

                               order_id order_status order_purchase_timestamp  \
0      00e7ee1b050b8499577073aeb2a297a1    delivered      2017-05-16 15:05:35   
1      29150127e6685892b6eab3eec79f59c7    delivered      2018-01-12 20:48:24   
2      b2059ed67ce144a36e2aa97d2c9e9ad2    delivered      2018-05-19 16:07:45   
3      951670f92359f4fe4a63112aa7306eba    delivered      2018-03-13 16:06:38   
4      6b7d50bd145f6fc7f33cebabd7e49d0f    delivered      2018-07-29 09:51:30   
...                                 ...          ...                      ...   
99436  6760e20addcf0121e9d58f2f1ff14298    delivered      2018-04-07 15:48:17   
99437  9ec0c8947d973db4f4e8dcf1fbfa8f1b    delivered      2018-04-04 08:20:22   
99438  fed4434add09a6f332ea398efd656a5c    delivered      2018-04-08 20:11:50   
99439  e31ec91cea1ecf97797787471f98a8c2    delivered      2017-11-03 21:08:33   
99440  28db69209a75e59f20ccbb5c36a20b90    delivered      2017-12-19 14:27:23   

         order_approved_at order_delivered_carrier_date  \
0      2017-05-16 15:22:12          2017-05-23 10:47:57   
1      2018-01-12 20:58:32          2018-01-15 17:14:59   
2      2018-05-20 16:19:10          2018-06-11 14:31:00   
3      2018-03-13 17:29:19          2018-03-27 23:22:42   
4      2018-07-29 10:10:09          2018-07-30 15:16:00   
...                    ...                          ...   
99436  2018-04-07 16:08:45          2018-04-11 02:08:36   
99437  2018-04-04 08:35:12          2018-04-05 18:42:35   
99438  2018-04-08 20:30:03          2018-04-09 17:52:17   
99439  2017-11-03 21:31:20          2017-11-06 18:24:41   
99440  2017-12-19 18:50:39          2017-12-21 19:17:21   

      order_delivered_customer_date order_estimated_delivery_date  
0               2017-05-25 10:35:35           2017-06-05 00:00:00  
1               2018-01-29 12:41:19           2018-02-06 00:00:00  
2               2018-06-14 17:58:51           2018-06-13 00:00:00  
3               2018-03-28 16:04:25           2018-04-10 00:00:00  
4               2018-08-09 20:55:48           2018-08-15 00:00:00  
...                             ...                           ...  
99436           2018-04-13 20:06:37           2018-04-25 00:00:00  


In [22]:
df_1 = df_0.merge(order_payments, on="order_id", how="inner")
df_1

customer_id                customer_unique_id  \
0       06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1       18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2       4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3       b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4       4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   
...                                  ...                               ...   
103881  17ddf5dd5d51696bb3d7c6291687be6f  1a29b476fee25c95fbafc67c5ac95cf8   
103882  e7b71a9017aa05c9a7fd292d714858e8  d52a67c98be1cf6a5c84435bd38d095d   
103883  5e28dfe12db7fb50a4b2f691faecea5e  e9f50caf99f032f0bf3c55141f019d99   
103884  56b18e2166679b8a959d72dd06da27f9  73c2643a0a458b49f58cea58833b192e   
103885  274fa6071e5e17fe303b9748641082c8  84732c5050c01db9b23e19ba39899398   

        customer_zip_code_prefix          customer_city customer_state  \
0                          14409                 franca             SP   
1                           9790  sao bernardo do campo             SP   
2                           1151              sao paulo             SP   
3                           8775        mogi das cruzes             SP   
4                          13056               campinas             SP   
...                          ...                    ...            ...   
103881                      3937              sao paulo             SP   
103882                      6764        taboao da serra             SP   
103883                     60115              fortaleza             CE   
103884                     92120                 canoas             RS   
103885                      6703                  cotia             SP   

                                order_id order_status  \
0       00e7ee1b050b8499577073aeb2a297a1    delivered   
1       29150127e6685892b6eab3eec79f59c7    delivered   
2       b2059ed67ce144a36e2aa97d2c9e9ad2    delivered   
3       951670f92359f4fe4a63112aa7306eba    delivered   
4       6b7d50bd145f6fc7f33cebabd7e49d0f    delivered   
...                                  ...          ...   
103881  6760e20addcf0121e9d58f2f1ff14298    delivered   
103882  9ec0c8947d973db4f4e8dcf1fbfa8f1b    delivered   
103883  fed4434add09a6f332ea398efd656a5c    delivered   
103884  e31ec91cea1ecf97797787471f98a8c2    delivered   
103885  28db69209a75e59f20ccbb5c36a20b90    delivered   

       order_purchase_timestamp    order_approved_at  \
0           2017-05-16 15:05:35  2017-05-16 15:22:12   
1           2018-01-12 20:48:24  2018-01-12 20:58:32   
2           2018-05-19 16:07:45  2018-05-20 16:19:10   
3           2018-03-13 16:06:38  2018-03-13 17:29:19   
4           2018-07-29 09:51:30  2018-07-29 10:10:09   
...                         ...                  ...   
103881      2018-04-07 15:48:17  2018-04-07 16:08:45   
103882      2018-04-04 08:20:22  2018-04-04 08:35:12   
103883      2018-04-08 20:11:50  2018-04-08 20:30:03   
103884      2017-11-03 21:08:33  2017-11-03 21:31:20   
103885      2017-12-19 14:27:23  2017-12-19 18:50:39   

       order_delivered_carrier_date order_delivered_customer_date  \
0               2017-05-23 10:47:57           2017-05-25 10:35:35   
1               2018-01-15 17:14:59           2018-01-29 12:41:19   
2               2018-06-11 14:31:00           2018-06-14 17:58:51   
3               2018-03-27 23:22:42           2018-03-28 16:04:25   
4               2018-07-30 15:16:00           2018-08-09 20:55:48   
...                             ...                           ...   
103881          2018-04-11 02:08:36           2018-04-13 20:06:37   
103882          2018-04-05 18:42:35           2018-04-11 18:54:45   
103883          2018-04-09 17:52:17           2018-05-09 19:03:15   
103884          2017-11-06 18:24:41           2017-11-16 19:58:39   
103885          2017-12-21 19:17:21           2017-12-26 18:42:36   

       order_est

Observamos que al unir ambos dataframes, ha aumentado el número de filas, por lo que puede suponer que existen valores duplicados en nuestra columna clave 'order_id'

In [23]:
# Encuentra los valores duplicados en la columna 'id_customer' de df1
duplicados_df0 = df_0[df_0['order_id'].duplicated(keep=False)]

# Encuentra los valores duplicados en la columna 'id_customer' de df2
duplicados_order_payments = order_payments[order_payments['order_id'].duplicated(keep=False)]

# Imprime la cantidad de duplicados en cada DataFrame
print("Duplicados en df_0:", len(duplicados_df0))
print("Duplicados en order_payments:", len(duplicados_order_payments))


Duplicados en df_0: 0
Duplicados en order_payments: 7407


Efectivamente, el dataframe 'order_payments' contiene 7407 valores duplicados. Esto puede debersea que se han utilizado varios métodos de pago en un mismo pedido. Vamos a comprobarlo:

In [24]:
duplicados_order_payments = order_payments[order_payments['order_id'].duplicated(keep=False)].sort_values(by='order_id')
duplicados_order_payments.head()

order_id  payment_sequential payment_type  \
80856  0016dfedd97fc2950e388d2971d718c7                   2      voucher   
89575  0016dfedd97fc2950e388d2971d718c7                   1  credit_card   
20036  002f19a65a2ddd70a090297872e6d64e                   1      voucher   
98894  002f19a65a2ddd70a090297872e6d64e                   2      voucher   
30155  0071ee2429bc1efdc43aa3e073a5290e                   2      voucher   

       payment_installments  payment_value  
80856                     1          17.92  
89575                     5          52.63  
20036                     1          44.11  
98894                     1          33.18  
30155                     1          92.44

Comprobamos que los pedidos duplicados se deben a que un cliente ha utilizado varios métodos de pago en un mismo pedido.

Para simplicar nuestro estudio, vamos a agrupar los pedidos duplicados en un único método de pago llamado "Varios"

In [25]:
df_agrupado = df_1.groupby('order_id').agg({'payment_type': ', '.join, 'payment_value': 'sum'}).reset_index()

In [26]:
df_agrupado['payment_type'] = df_agrupado['payment_type'].apply(lambda x: 'varios' if ', ' in x else x)

In [27]:
df_agrupado.head()

order_id payment_type  payment_value
0  00010242fe8c5a6d1ba2dd792cb16214  credit_card          72.19
1  00018f77f2f0320c557190d7a144bdd3  credit_card         259.83
2  000229ec398224ef6ca0657da4fc703e  credit_card         216.87
3  00024acbcdf0a6daa1e931b038114c75  credit_card          25.78
4  00042b26cf59d7ce69dfabb4e55b4fd9  credit_card         218.04

In [28]:
df_1[df_1['order_id'] == '0016dfedd97fc2950e388d2971d718c7']

customer_id                customer_unique_id  \
76103  2c8b917c5d7dd720ebe36a5ed3b501ec  33febb33a42a35f5826a22584b127f9d   
76104  2c8b917c5d7dd720ebe36a5ed3b501ec  33febb33a42a35f5826a22584b127f9d   

       customer_zip_code_prefix customer_city customer_state  \
76103                     45860   canavieiras             BA   
76104                     45860   canavieiras             BA   

                               order_id order_status order_purchase_timestamp  \
76103  0016dfedd97fc2950e388d2971d718c7    delivered      2017-04-28 19:54:40   
76104  0016dfedd97fc2950e388d2971d718c7    delivered      2017-04-28 19:54:40   

         order_approved_at order_delivered_carrier_date  \
76103  2017-04-29 10:05:12          2017-05-16 08:40:49   
76104  2017-04-29 10:05:12          2017-05-16 08:40:49   

      order_delivered_customer_date order_estimated_delivery_date  \
76103           2017-05-22 18:32:21           2017-06-01 00:00:00   
76104           2017-05-22 18:32:21           2017-06-01 00:00:00   

       payment_sequential payment_type  payment_installments  payment_value  
76103                   2      voucher                     1          17.92  
76104                   1  credit_card                     5          52.63

In [29]:
df_agrupado[df_agrupado['order_id'] == '0016dfedd97fc2950e388d2971d718c7']

order_id payment_type  payment_value
36  0016dfedd97fc2950e388d2971d718c7       varios          70.55

Tras estas operaciones, hemos conseguido fusionar aquellos pedidos con varios métodos de pago en un método de pago común llamado "varios", además de sumar el total del valor de pago.

In [30]:
df_2 = df_agrupado.merge(df_0, on="order_id", how="inner")
df_2

order_id payment_type  payment_value  \
0      00010242fe8c5a6d1ba2dd792cb16214  credit_card          72.19   
1      00018f77f2f0320c557190d7a144bdd3  credit_card         259.83   
2      000229ec398224ef6ca0657da4fc703e  credit_card         216.87   
3      00024acbcdf0a6daa1e931b038114c75  credit_card          25.78   
4      00042b26cf59d7ce69dfabb4e55b4fd9  credit_card         218.04   
...                                 ...          ...            ...   
99435  fffc94f6ce00a00581880bf54a75a037       boleto         343.40   
99436  fffcd46ef2263f404302a634eb57f7eb       boleto         386.53   
99437  fffce4705a9662cd70adb13d4a31832d  credit_card         116.85   
99438  fffe18544ffabc95dfada21779c9644f  credit_card          64.71   
99439  fffe41c64501cc87c801fd61db3f6244  credit_card          55.79   

                            customer_id                customer_unique_id  \
0      3ce436f183e68e07877b285a838db11a  871766c5855e863f6eccc05f988b23cb   
1      f6dd3ec061db4e3987629fe6b26e5cce  eb28e67c4c0b83846050ddfb8a35d051   
2      6489ae5e4333f3693df5ad4372dab6d3  3818d81c6709e39d06b2738a8d3a2474   
3      d4eb9395c8c0431ee92fce09860c5a06  af861d436cfc08b2c2ddefd0ba074622   
4      58dbd0b2d70206bf40e62cd34e84d795  64b576fb70d441e8f1b2d7d446e483c5   
...                                 ...                               ...   
99435  b51593916b4b8e0d6f66f2ae24f2673d  0c9aeda10a71f369396d0c04dce13a64   
99436  84c5d4fbaf120aae381fad077416eaa0  0da9fe112eae0c74d3ba1fe16de0988b   
99437  29309aa813182aaddc9b259e31b870e6  cd79b407828f02fdbba457111c38e4c4   
99438  b5e6afd5a41800fdf401e0272ca74655  eb803377c9315b564bdedad672039306   
99439  96d649da0cc4ff33bb408b199d4c7dcf  cd76a00d8e3ca5e6ab9ed9ecb6667ac4   

       customer_zip_code_prefix          customer_city customer_state  \
0                         28013  campos dos goytacazes             RJ   
1                         15775        santa fe do sul             SP   
2                         35661          para de minas             MG   
3                         12952                atibaia             SP   
4                         13226        varzea paulista             SP   
...                         ...                    ...            ...   
99435                     65077               sao luis             MA   
99436                     81690               curitiba             PR   
99437                      4039              sao paulo             SP   
99438                     13289                vinhedo             SP   
99439                     18605               botucatu             SP   

      order_status order_purchase_timestamp    order_approved_at  \
0        delivered      2017-09-13 08:59:02  2017-09-13 09:45:35   
1        delivered      2017-04-26 10:53:06  2017-04-26 11:05:13   
2        delivered      2018-01-14 14:33:31  2018-01-14 14:48:30   
3        delivered      2018-08-08 10:00:35  2018-08-08 10:10:18   
4        delivered      2017-02-04 13:57:51  2017-02-04 14:10:13   
...            ...                      ...                  ...   
99435    delivered      2018-04-23 13:57:06  2018-04-25 04:11:01   
99436    delivered      2018-07-14 10:26:46  2018-07-17 04:31:48   
99437    delivered      2017-10-23 17:07:56  2017-10-24 17:14:25   
99438    delivered      2017-08-14 23:02:59  2017-08-15 00:04:32   
99439    delivered      2018-06-09 17:00:18  2018-06-09 17:10:13   

      order_delivered_carrier_date order_delivered_customer_date  \
0              2017-09-19 18:34:16           2017-09-20 23:43:48   
1              2017-05-04 14:35:00           2017-05-12 16:04:24   
2              2018-01-16 12:36:48           2018-01-22 13:19:16   
3              2018-08-10 13:28:00           2018-08-14 13:32:39   
4              2017-02-16 09:46:09           2017-03-01 16:42:31   
...                            ...                           ...   
99435          2018-04-25 12:09:00           2018-05-10 22:56:40   
99436          

In [31]:
df_3 = df_2.merge(order_items, on="order_id", how="inner")
df_3

order_id payment_type  payment_value  \
0       00010242fe8c5a6d1ba2dd792cb16214  credit_card          72.19   
1       00018f77f2f0320c557190d7a144bdd3  credit_card         259.83   
2       000229ec398224ef6ca0657da4fc703e  credit_card         216.87   
3       00024acbcdf0a6daa1e931b038114c75  credit_card          25.78   
4       00042b26cf59d7ce69dfabb4e55b4fd9  credit_card         218.04   
...                                  ...          ...            ...   
112642  fffc94f6ce00a00581880bf54a75a037       boleto         343.40   
112643  fffcd46ef2263f404302a634eb57f7eb       boleto         386.53   
112644  fffce4705a9662cd70adb13d4a31832d  credit_card         116.85   
112645  fffe18544ffabc95dfada21779c9644f  credit_card          64.71   
112646  fffe41c64501cc87c801fd61db3f6244  credit_card          55.79   

                             customer_id                customer_unique_id  \
0       3ce436f183e68e07877b285a838db11a  871766c5855e863f6eccc05f988b23cb   
1       f6dd3ec061db4e3987629fe6b26e5cce  eb28e67c4c0b83846050ddfb8a35d051   
2       6489ae5e4333f3693df5ad4372dab6d3  3818d81c6709e39d06b2738a8d3a2474   
3       d4eb9395c8c0431ee92fce09860c5a06  af861d436cfc08b2c2ddefd0ba074622   
4       58dbd0b2d70206bf40e62cd34e84d795  64b576fb70d441e8f1b2d7d446e483c5   
...                                  ...                               ...   
112642  b51593916b4b8e0d6f66f2ae24f2673d  0c9aeda10a71f369396d0c04dce13a64   
112643  84c5d4fbaf120aae381fad077416eaa0  0da9fe112eae0c74d3ba1fe16de0988b   
112644  29309aa813182aaddc9b259e31b870e6  cd79b407828f02fdbba457111c38e4c4   
112645  b5e6afd5a41800fdf401e0272ca74655  eb803377c9315b564bdedad672039306   
112646  96d649da0cc4ff33bb408b199d4c7dcf  cd76a00d8e3ca5e6ab9ed9ecb6667ac4   

        customer_zip_code_prefix          customer_city customer_state  \
0                          28013  campos dos goytacazes             RJ   
1                          15775        santa fe do sul             SP   
2                          35661          para de minas             MG   
3                          12952                atibaia             SP   
4                          13226        varzea paulista             SP   
...                          ...                    ...            ...   
112642                     65077               sao luis             MA   
112643                     81690               curitiba             PR   
112644                      4039              sao paulo             SP   
112645                     13289                vinhedo             SP   
112646                     18605               botucatu             SP   

       order_status order_purchase_timestamp    order_approved_at  \
0         delivered      2017-09-13 08:59:02  2017-09-13 09:45:35   
1         delivered      2017-04-26 10:53:06  2017-04-26 11:05:13   
2         delivered      2018-01-14 14:33:31  2018-01-14 14:48:30   
3         delivered      2018-08-08 10:00:35  2018-08-08 10:10:18   
4         delivered      2017-02-04 13:57:51  2017-02-04 14:10:13   
...             ...                      ...                  ...   
112642    delivered      2018-04-23 13:57:06  2018-04-25 04:11:01   
112643    delivered      2018-07-14 10:26:46  2018-07-17 04:31:48   
112644    delivered      2017-10-23 17:07:56  2017-10-24 17:14:25   
112645    delivered      2017-08-14 23:02:59  2017-08-15 00:04:32   
112646    delivered      2018-06-09 17:00:18  2018-06-09 17:10:13   

       order_delivered_carrier_date order_delivered_customer_date  \
0               2017-09-19 18:34:16           2017-09-20 23:43:48   
1               2017-05-04 14:35:00           2017-05-12 16:04:24   
2               2018-01-16 12:36:48           2018-01-22 13:19:16   
3               2018-08-10 13:28:00           2018-08-14 13:32:39   
4               2017-02-16 09:46:09           2017-03-01 16:42:31   
...                             ...                           ...   
112642          2018-04-25 12

Al unir con el dataframe de order_items volvemos a tener un problema similar al de antes. Hay order_id duplicadas ya que un mismo pedido puede contener distintos productos. Hacemos la comprobación.

In [32]:
# Encuentra los valores duplicados en la columna 'order_id' de df_3
duplicados_df3 = df_3[df_3['order_id'].duplicated(keep=False)]

# Imprime la cantidad de duplicados en el DataFrame
print("Duplicados en df_3:", len(duplicados_df3))

Duplicados en df_3: 23784


In [33]:
duplicados_df3.sort_values(by='order_id')

order_id payment_type  payment_value  \
13      0008288aa423d2a3f00fcb17cd7d8719       boleto         126.54   
14      0008288aa423d2a3f00fcb17cd7d8719       boleto         126.54   
32      00143d0f86d6fbd9f9b38ab440ac16f5  credit_card         109.29   
33      00143d0f86d6fbd9f9b38ab440ac16f5  credit_card         109.29   
34      00143d0f86d6fbd9f9b38ab440ac16f5  credit_card         109.29   
...                                  ...          ...            ...   
112631  fff8287bbae429a99bb7e8c21d151c41  credit_card         456.28   
112638  fffb9224b6fc7c43ebb0904318b10b5f       boleto         356.76   
112639  fffb9224b6fc7c43ebb0904318b10b5f       boleto         356.76   
112637  fffb9224b6fc7c43ebb0904318b10b5f       boleto         356.76   
112640  fffb9224b6fc7c43ebb0904318b10b5f       boleto         356.76   

                             customer_id                customer_unique_id  \
13      2355af7c75e7c98b43a87b2a7f210dc5  9e4159995424971423b98c4a8bc11529   
14      2355af7c75e7c98b43a87b2a7f210dc5  9e4159995424971423b98c4a8bc11529   
32      2e45292979b9b2700ea09560eeb0f803  bb8a37225e0279ac8a274c9765617eaf   
33      2e45292979b9b2700ea09560eeb0f803  bb8a37225e0279ac8a274c9765617eaf   
34      2e45292979b9b2700ea09560eeb0f803  bb8a37225e0279ac8a274c9765617eaf   
...                                  ...                               ...   
112631  6c1e92a209dbf868706caa831090941e  028c09f007292c4e3a3b10d296e47987   
112638  4d3abb73ceb86353aeadbe698aa9d5cb  f736308cd9952b33b90b9fe94da9c8f5   
112639  4d3abb73ceb86353aeadbe698aa9d5cb  f736308cd9952b33b90b9fe94da9c8f5   
112637  4d3abb73ceb86353aeadbe698aa9d5cb  f736308cd9952b33b90b9fe94da9c8f5   
112640  4d3abb73ceb86353aeadbe698aa9d5cb  f736308cd9952b33b90b9fe94da9c8f5   

        customer_zip_code_prefix  customer_city customer_state order_status  \
13                          6600        jandira             SP    delivered   
14                          6600        jandira             SP    delivered   
32                         74663        goiania             GO    delivered   
33                         74663        goiania             GO    delivered   
34                         74663        goiania             GO    delivered   
...                          ...            ...            ...          ...   
112631                     58075    joao pessoa             PB    delivered   
112638                     56912  serra talhada             PE    delivered   
112639                     56912  serra talhada             PE    delivered   
112637                     56912  serra talhada             PE    delivered   
112640                     56912  serra talhada             PE    delivered   

       order_purchase_timestamp    order_approved_at  \
13          2018-02-13 22:10:21  2018-02-15 03:55:52   
14          2018-02-13 22:10:21  2018-02-15 03:55:52   
32          2017-10-16 15:29:43  2017-10-16 16:07:52   
33          2017-10-16 15:29:43  2017-10-16 16:07:52   
34          2017-10-16 15:29:43  2017-10-16 16:07:52   
...                         ...                  ...   
112631      2018-03-17 12:11:45  2018-03-17 12:29:22   
112638      2017-10-27 16:51:00  2017-10-28 02:55:58   
112639      2017-10-27 16:51:00  2017-10-28 02:55:58   
112637      2017-10-27 16:51:00  2017-10-28 02:55:58   
112640      2017-10-27 16:51:00  2017-10-28 02:55:58   

       order_delivered_carrier_date order_delivered_customer_date  \
13              2018-02-20 18:23:28           2018-02-26 13:55:22   
14              2018-02-20 18:23:28           2018-02-26 13:55:22   
32              2017-10-16 23:04:43           2017-10-27 18:43:56   
33              2017-10-16 23:04:43           2017-10-27 18:43:56   
34              2017-10-16 23:04:43           2017-10-27 18:43:56   
...                             ...                           ...   
112631          2018-03-21 19:22:25           2018-04-07 10:07:48   
112638          2017-11-10 19:31:52           2017-11-17

Observamos que se trata de pedidos con varias unidades de un mismo artículo o con varias unidades de distintos artículos.
Podemos optar por mantener estos pedidos duplicados y tratar de aprovechar los datos de información de los productos de cara a la segmentación.

A continuación, continuamos fusionando con los otros dataframes.

In [34]:
df_4 = df_3.merge(products, on="product_id", how="inner")
df_4

order_id payment_type  payment_value  \
0       00010242fe8c5a6d1ba2dd792cb16214  credit_card          72.19   
1       130898c0987d1801452a8ed92a670612       boleto          73.86   
2       532ed5e14e24ae1f0d735b91524b98b9  credit_card          83.23   
3       6f8c31653edb8c83e1a739408b5ff750  credit_card          75.07   
4       7d19f4ef4d04461989632411b7e588b9  credit_card          72.19   
...                                  ...          ...            ...   
112642  ffebd80e3291e811c308365936897efd  credit_card         181.10   
112643  ffee31fb4b5e35c9123608015637c495  credit_card         112.52   
112644  fff7c4452f050315db1b3f24d9df5fcd  credit_card         756.91   
112645  fffa82886406ccf10c7b4e35c4ff2788  credit_card         273.92   
112646  fffe41c64501cc87c801fd61db3f6244  credit_card          55.79   

                             customer_id                customer_unique_id  \
0       3ce436f183e68e07877b285a838db11a  871766c5855e863f6eccc05f988b23cb   
1       e6eecc5a77de221464d1c4eaff0a9b64  0fb8e3eab2d3e79d92bb3fffbb97f188   
2       4ef55bf80f711b372afebcb7c715344a  3419052c8c6b45daf79c1e426f9e9bcb   
3       30407a72ad8b3f4df4d15369126b20c9  e7c828d22c0682c1565252deefbe334d   
4       91a792fef70ecd8cc69d3c7feb3d12da  0bb98ba72dcc08e95f9d8cc434e9a2cc   
...                                  ...                               ...   
112642  050309b91cc5e04e68841938e7984aaf  a1c2e4b4c1b29e41684ebdf3cc6538f8   
112643  11e0f43ab4e2d2c48348dd9332c0ef80  4e15fee963523774998876e3624952f4   
112644  dec8952e97ef6124259c56914fb3569c  c86ab5ff2bac777c4e770a383feca4eb   
112645  a5201e1a6d71a8d21e869151bd5b4085  2a3ab9bf9639491997586882c502540a   
112646  96d649da0cc4ff33bb408b199d4c7dcf  cd76a00d8e3ca5e6ab9ed9ecb6667ac4   

        customer_zip_code_prefix          customer_city customer_state  \
0                          28013  campos dos goytacazes             RJ   
1                          75800                  jatai             GO   
2                          30720         belo horizonte             MG   
3                          83070   sao jose dos pinhais             PR   
4                          36400   conselheiro lafaiete             MG   
...                          ...                    ...            ...   
112642                     28740    conceicao de macabu             RJ   
112643                     83065   sao jose dos pinhais             PR   
112644                     27261          volta redonda             RJ   
112645                     59955        tenente ananias             RN   
112646                     18605               botucatu             SP   

       order_status order_purchase_timestamp  ...  price freight_value  \
0         delivered      2017-09-13 08:59:02  ...   58.9         13.29   
1         delivered      2017-06-28 11:52:20  ...   55.9         17.96   
2         delivered      2018-05-18 10:25:53  ...   64.9         18.33   
3         delivered      2017-08-01 18:38:42  ...   58.9         16.17   
4         delivered      2017-08-10 21:48:40  ...   58.9         13.29   
...             ...                      ...  ...    ...           ...   
112642    delivered      2018-07-13 20:04:05  ...  129.9         51.20   
112643    delivered      2018-08-18 10:00:59  ...   99.0         13.52   
112644    delivered      2017-06-01 16:53:03  ...  736.0         20.91   
112645    delivered      2017-12-18 16:33:07  ...  229.9         44.02   
112646    delivered      2018-06-09 17:00:18  ...   43.0         12.79   

       product_category_name product_name_lenght  product_description_lenght  \
0                 cool_stuff                58.0                       598.0   
1                 cool_stuff                58.0                       598.0   
2                 cool_stuff                58.0                       598.0   
3                 cool_stuff                58.0                       598.0   
4                 cool_stuff                58.0    

In [35]:
df_5 = df_4.merge(order_reviews, on="order_id", how="inner")
df_5

order_id payment_type  payment_value  \
0       00010242fe8c5a6d1ba2dd792cb16214  credit_card          72.19   
1       130898c0987d1801452a8ed92a670612       boleto          73.86   
2       532ed5e14e24ae1f0d735b91524b98b9  credit_card          83.23   
3       6f8c31653edb8c83e1a739408b5ff750  credit_card          75.07   
4       7d19f4ef4d04461989632411b7e588b9  credit_card          72.19   
...                                  ...          ...            ...   
112364  ffebd80e3291e811c308365936897efd  credit_card         181.10   
112365  ffee31fb4b5e35c9123608015637c495  credit_card         112.52   
112366  fff7c4452f050315db1b3f24d9df5fcd  credit_card         756.91   
112367  fffa82886406ccf10c7b4e35c4ff2788  credit_card         273.92   
112368  fffe41c64501cc87c801fd61db3f6244  credit_card          55.79   

                             customer_id                customer_unique_id  \
0       3ce436f183e68e07877b285a838db11a  871766c5855e863f6eccc05f988b23cb   
1       e6eecc5a77de221464d1c4eaff0a9b64  0fb8e3eab2d3e79d92bb3fffbb97f188   
2       4ef55bf80f711b372afebcb7c715344a  3419052c8c6b45daf79c1e426f9e9bcb   
3       30407a72ad8b3f4df4d15369126b20c9  e7c828d22c0682c1565252deefbe334d   
4       91a792fef70ecd8cc69d3c7feb3d12da  0bb98ba72dcc08e95f9d8cc434e9a2cc   
...                                  ...                               ...   
112364  050309b91cc5e04e68841938e7984aaf  a1c2e4b4c1b29e41684ebdf3cc6538f8   
112365  11e0f43ab4e2d2c48348dd9332c0ef80  4e15fee963523774998876e3624952f4   
112366  dec8952e97ef6124259c56914fb3569c  c86ab5ff2bac777c4e770a383feca4eb   
112367  a5201e1a6d71a8d21e869151bd5b4085  2a3ab9bf9639491997586882c502540a   
112368  96d649da0cc4ff33bb408b199d4c7dcf  cd76a00d8e3ca5e6ab9ed9ecb6667ac4   

        customer_zip_code_prefix          customer_city customer_state  \
0                          28013  campos dos goytacazes             RJ   
1                          75800                  jatai             GO   
2                          30720         belo horizonte             MG   
3                          83070   sao jose dos pinhais             PR   
4                          36400   conselheiro lafaiete             MG   
...                          ...                    ...            ...   
112364                     28740    conceicao de macabu             RJ   
112365                     83065   sao jose dos pinhais             PR   
112366                     27261          volta redonda             RJ   
112367                     59955        tenente ananias             RN   
112368                     18605               botucatu             SP   

       order_status order_purchase_timestamp  ... product_weight_g  \
0         delivered      2017-09-13 08:59:02  ...            650.0   
1         delivered      2017-06-28 11:52:20  ...            650.0   
2         delivered      2018-05-18 10:25:53  ...            650.0   
3         delivered      2017-08-01 18:38:42  ...            650.0   
4         delivered      2017-08-10 21:48:40  ...            650.0   
...             ...                      ...  ...              ...   
112364    delivered      2018-07-13 20:04:05  ...           6700.0   
112365    delivered      2018-08-18 10:00:59  ...           2300.0   
112366    delivered      2017-06-01 16:53:03  ...            400.0   
112367    delivered      2017-12-18 16:33:07  ...           2700.0   
112368    delivered      2018-06-09 17:00:18  ...            600.0   

       product_length_cm product_height_cm product_width_cm  \
0                   28.0               9.0             14.0   
1                   28.0               9.0             14.0   
2                   28.0               9.0             14.0   
3                   28.0               9.0             14.0   
4                   28.0               9.0             14.0   
...                  ...               ...              ...   
112364              35.0              12.0             22.0

In [36]:
df_inicial = df_5[['customer_id','customer_zip_code_prefix','customer_city','customer_state','order_id','price','freight_value','payment_type','payment_value','product_id','product_category_name','order_status','order_purchase_timestamp','order_approved_at','review_id','review_score']]
df_inicial

customer_id  customer_zip_code_prefix  \
0       3ce436f183e68e07877b285a838db11a                     28013   
1       e6eecc5a77de221464d1c4eaff0a9b64                     75800   
2       4ef55bf80f711b372afebcb7c715344a                     30720   
3       30407a72ad8b3f4df4d15369126b20c9                     83070   
4       91a792fef70ecd8cc69d3c7feb3d12da                     36400   
...                                  ...                       ...   
112364  050309b91cc5e04e68841938e7984aaf                     28740   
112365  11e0f43ab4e2d2c48348dd9332c0ef80                     83065   
112366  dec8952e97ef6124259c56914fb3569c                     27261   
112367  a5201e1a6d71a8d21e869151bd5b4085                     59955   
112368  96d649da0cc4ff33bb408b199d4c7dcf                     18605   

                customer_city customer_state  \
0       campos dos goytacazes             RJ   
1                       jatai             GO   
2              belo horizonte             MG   
3        sao jose dos pinhais             PR   
4        conselheiro lafaiete             MG   
...                       ...            ...   
112364    conceicao de macabu             RJ   
112365   sao jose dos pinhais             PR   
112366          volta redonda             RJ   
112367        tenente ananias             RN   
112368               botucatu             SP   

                                order_id  price  freight_value payment_type  \
0       00010242fe8c5a6d1ba2dd792cb16214   58.9          13.29  credit_card   
1       130898c0987d1801452a8ed92a670612   55.9          17.96       boleto   
2       532ed5e14e24ae1f0d735b91524b98b9   64.9          18.33  credit_card   
3       6f8c31653edb8c83e1a739408b5ff750   58.9          16.17  credit_card   
4       7d19f4ef4d04461989632411b7e588b9   58.9          13.29  credit_card   
...                                  ...    ...            ...          ...   
112364  ffebd80e3291e811c308365936897efd  129.9          51.20  credit_card   
112365  ffee31fb4b5e35c9123608015637c495   99.0          13.52  credit_card   
112366  fff7c4452f050315db1b3f24d9df5fcd  736.0          20.91  credit_card   
112367  fffa82886406ccf10c7b4e35c4ff2788  229.9          44.02  credit_card   
112368  fffe41c64501cc87c801fd61db3f6244   43.0          12.79  credit_card   

        payment_value                        product_id product_category_name  \
0               72.19  4244733e06e7ecb4970a6e2683c13e61            cool_stuff   
1               73.86  4244733e06e7ecb4970a6e2683c13e61            cool_stuff   
2               83.23  4244733e06e7ecb4970a6e2683c13e61            cool_stuff   
3               75.07  4244733e06e7ecb4970a6e2683c13e61            cool_stuff   
4               72.19  4244733e06e7ecb4970a6e2683c13e61            cool_stuff   
...               ...                               ...                   ...   
112364         181.10  4cc4d02efc8f249c13355147fb44e34d    ferramentas_jardim   
112365         112.52  b10ecf8e33aaaea419a9fa860ea80fb5      moveis_decoracao   
112366         756.91  dd469c03ad67e201bc2179ef077dcd48    relogios_presentes   
112367         273.92  bbe7651fef80287a816ead73f065fc4b         esporte_lazer   
112368          55.79  350688d9dc1e75ff97be326363655e01       cama_mesa_banho   

       order_status order_purchase_timestamp    order_approved_at  \
0         delivered      2017-09-13 08:59:02  2017-09-13 09:45:35   
1         delivered      2017-06-28 11:52:20  2017-06-29 02:44:11   
2         delivered      2018-05-18 10:25:53  2018-05-18 12:31:43   
3         delivered      2017-08-01 18:38:42  2017-08-01 18:55:08   
4         delivered      2017-08-10 21:48:40  2017-08-10 22:05:11   
...             ...                      ...                  ...   
112364    delivered      2018-07-13 20:04:05  2018-07-13 20:23:55   
112365    delivered      2018-08-18 10:00:59  2018-08-18 10:10:11   
112366    delivered      2017-06-01 16:53:03  2017-06-01 17:05:23   

Guardamos en csv

In [37]:
df_inicial.to_csv('df_inicial.csv', index=False)